tyrian/src/main/scala/tyrian/http/Models.scala

```diff
@@ -41,6 +43,7 @@ enum Body derives CanEqual:
     *   the content of the body
     */
   case PlainText(contentType: String, body: String) extends Body
+  case File(contentType: String, body: BodyInit) extends Body
```

Comment:
This might be ok, but I'm a bit worried about using BodyInit as a catch-all type. I am entirely open to counter arguments. 🙂 

I haven't tried this so this is speculation, but from reading around I think we need something like this (though this seems cumbersome):

```scala
enum Body derives CanEqual:
  case Empty
  case PlainText(contentType: String, body: String)
  case File(contentType: String, body: org.scalajs.dom.File)
  case Blob(contentType: String, body: org.scalajs.dom.Blob)
  case FormData(contentType: String, body: org.scalajs.dom.FormData)
  case ArrayBuffer(contentType: String, body: org.scalajs.dom.ArrayBuffer)
  case TypedArray(contentType: String, body: org.scalajs.dom.TypedArray)
  case URLSearchParams(contentType: String, body: org.scalajs.dom.URLSearchParams)
  case ReadableStream(contentType: String, body: org.scalajs.dom.ReadableStream)
```

It also feels like we could do with an enum of content types, and then we could provide helper methods so that you could do things like:

```scala
def bodyImagePng(data: Blob): Body =
  Body.Blob(ContentType.imagePng, data)
```

Or something. What do you think?

Reply:
A list of content types is a [big 'ol list](https://developer.mozilla.org/en-US/docs/Web/HTTP/MIME_types/Common_types) - maybe just types we support? I suspect for now you probably just want `PlainText`, `File`, and `Blob` to start.. dealing with `application/json` and `application/xml` are problems all on their own for another time :slightly_smiling_face:

Reply:
Possibly crossed wires (or I'm just wrong, wouldn't be the first time 😁 ), we're not talking mime-types, we're talking bodyinit types - I think that's the whole list ☝️.

References:
https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/send#parameters
https://fetch.spec.whatwg.org/#bodyinit-unions

> An XMLHttpRequestBodyInit, which per the Fetch spec can be a Blob, an ArrayBuffer, a TypedArray, a DataView, a FormData, a URLSearchParams, or a string.

Not...sure...where I got `ReadableStream` from... 🤔 

Does that change anything for you?